.github/workflows/node.js.yaml

```diff
@@ -0,0 +1,35 @@
+name: Node.js CI
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x, 18.x]
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: "npm"
+      - run: npm ci
+      - run: npm run build --if-present
+      - run: npm test
+      - name: Prettier check
+        uses: creyD/prettier_action@v4.2
+        with:
+          prettier_options: --check **/*.{ts,md}
```

Comment:
We have more file types than these two (e.g., `tsx`).  Let's make sure this includes them all.